how-to/how-to-use-kata-containers-with-nemu.md

```diff
@@ -0,0 +1,59 @@
+# Kata Containers with NEMU
+Kata Containers relies by default on Qemu hypervisor in order to spawn the virtual machines running containers. [NEMU](https://github.com/intel/nemu) is a fork of Qemu focused on increasing the security through 3 different ways:
+- Reducing the number of lines of code.
+- Removing all legacy devices.
+- Reducing the emulation as far as possible.
+
+As the goal for Kata Containers is to increase the security, this is a logical step to get Kata relying on NEMU as its hypervisor.
+
+## How to get started
+
+This section will describe how to run Kata Containers with NEMU, first by explaining how to download, build and install it. Then this document will walk through the steps needed to update your Kata Containers configuration in order to run with NEMU. 
+
+### Pre-requisites
+This document requires Kata Containers to be installed in your system.
+
+### NEMU
+___Download, build and install___
+```bash
+$ git clone https://github.com/intel/nemu.git
+$ cd nemu
+$ git fetch origin/master
+$ git checkout experiment/automatic-removal
+$ SRCDIR=$PWD ./tools/build_x86_64_virt.sh
+```
+After those commands successfully returned, you will find the NEMU binary at `$HOME/build-x86_64/x86_64_virt-softmmu/qemu-system-x86_64_virt`.
+
+You also need the `OVMF` firmware in order to boot the virtual machine's kernel. It can currently be found at this [location](https://github.com/rbradford/edk2/releases/). Please choose the latest version of this firmware (`v0.2` in this example):
+```bash
+$ mkdir -p $HOME/nemu_firmware
+$ wget https://github.com/rbradford/edk2/releases/download/0.2/OVMF.fd -P $HOME/nemu_firmware/
+```
+
+### Configure Kata Containers
+All you need from this section is to modify the configuration file `configuration.toml`, to specify the options related to the hypervisor.
+___`/usr/share/defaults/kata-containers/configuration.toml`___
+```toml
+[hypervisor.qemu]
+path = "/home/foo/build-x86_64/x86_64_virt-softmmu/qemu-system-x86_64_virt"
+...
+machine_type = "virt"
+...
+firmware = "/home/foo/nemu_firmware/OVMF.fd"
+...
+```
+As you can see from this snippet above, all you need to change is:
+- The path to the hypervisor binary, `/home/foo/build-x86_64/x86_64_virt-softmmu/qemu-system-x86_64_virt` in this example.
+- The machine type from `pc` to `virt`.
+- The path to the firmware binary, `/home/foo/nemu_firmware/OVMF.fd` in this example.
+
+Once you have saved those modifications, you can start a new container:
+```bash
+$ docker run --runtime=kata-runtime -it busybox
+```
+And you will be able to verify this new container is running with NEMU hypervisor by looking for the hypervisor path and the machine type from the `qemu` process running on your system:
```

Comment:
- s/with NEMU/with the NEMU/

- What about running `kata-runtime kata-env`?

  I think we should ensure that it is obvious from that output that the runtime is configured to use a NEMU hypervisor so if not already done, we should patch the output of `--version` to include "`nemu`".

  Related to this comment (but not related to this PR), is there anything else we need to do https://github.com/kata-containers/runtime/blob/master/data/kata-collect-data.sh.in for NEMU?

Reply:
Could you open an issue on NEMU repo for that, I think this would be a great addition (`--version` printing out NEMU somewhere).

Reply:
Done - see https://github.com/intel/nemu/issues/127.